### ISSUE TITLE
Assemble/find a list of test vectors

### ISSUE THREAD

**Issue by est31** [RustAudio/mp3#7]
We need some mp3 files to test the code on. They need to be licensed appropriately, e.g. public domain. They should reflect a variety of encoders as well as encoder parameters, modes, etc.

This is a good starting point, but I don't know which format the .bit files are in: https://github.com/lieff/minimp3/tree/master/vectors

**Comment by mainrs** [comment]
Mixed data. Some are layers, some are valid mp3 files. Others only contain the header.

**Comment by lieff** [comment]
Good point. I've added self-generated vectors which I use for performance testing https://github.com/lieff/minimp3/tree/master/vectors/performance . This is public domain.
As for others, most of them from testing procedure supplied with ISO standard and reference code. Unfortunately my old archive do not contain any license file. ffmpeg fate uses same standard vectors http://fate-suite.ffmpeg.org/mp3-conformance/ and no license information too. I want find someone who clarify what license for standard reference vectors. But currently I just store them same way as ffmpeg fate.

Others based on standard vectors. I'm not sure about ILL_* origin (this is specially-broken vectors for stability\error correction testing), I need time to investigate.

**Comment by lieff** [comment]
About .bit files format:
ILL* - mixed layer 1\2\3
l1_* - MPEG layer 1
l2_* - MPEG layer 2
l3_* - MPEG layer 3
M2L3_* MPEG layer 3
MPEG versions also mixed.

**Comment by mainrs** [comment]
@lieff how did you find that out? I tried running file on them but it returned garbage for me. :)

**Comment by lieff** [comment]
@SirWindfield For ILL* and \*nonstandard* it's normal, they broken to trigger bugs in decoder. Others should be decoded\played normally.
ffplay can play them, (but showed layer 1 and layer 2 as mp3 too).

**Comment by mainrs** [comment]
I was wondering, the RustAudio group, is it also "responsible" for metadata? What about merging existing projects into the organization, like rust-id3 and rust-metaflac?

**Comment by est31** [comment]
@SirWindfield existing projects of these kinds are definitely welcome!

**Comment by lieff** [comment]
ILL* vectors is from Philips MPEG Audio Test Suite http://www.mpegmultichannel.free.fr/Mpeg.pdf

**Comment by lieff** [comment]
Here also https://samples.ffmpeg.org/A-codecs/MP3/ no mention of license too.

**Comment by mainrs** [comment]
@lieff I found this some directories up in the tree: https://samples.ffmpeg.org/HEADER.txt.

**Comment by lieff** [comment]
So, licence should be clarified at webmaster(AT)ffmpeg(DOT)org.
